runtime/executor/src/main/java/edu/snu/onyx/runtime/executor/data/DataUtil.java

```
@@ -155,4 +158,46 @@ public static Iterable concatNonSerPartitions(final Iterable<NonSerializedPartit
     }
     return concatStream.collect(Collectors.toList());
   }
+
+  /**
+   * Creates new OutputStream which writes compressed data onto the stream.
+   *
+   * @param out the original {@link OutputStream}.
+   * @param compressor compress strategy for compressing stream data.
+   * @return {@link OutputStream} which writes compressed data.
+   * @throws IOException if fail to compress data.
+   */
+  OutputStream createOutputStream(final OutputStream out, final Compressor compressor) throws IOException {
+    switch (compressor) {
+      case Raw:
+        return new BufferedOutputStream(out);
+      case Gzip:
+        return new GZIPOutputStream(out);
+      case LZ4:
+        return null; // add later (maybe adding dependency?)
+      default:
+        return null;
```

You have to throw an exception instead of `null`.